tests/manifests/__init__.py

```diff
@@ -111,6 +113,11 @@ def __post_init__(self):
             self.has_service_monitor = self.has_workloads
         if self.has_topology_spread_constraints is None:
             self.has_topology_spread_constraints = self.has_workloads
+        self.has_replicas = self.has_workloads and (
+            not self.values_file_path_overrides
+            or self.values_file_path_overrides
+            and self.values_file_path_overrides.get(PropertyType.Replicas) != ValuesFilePath.not_supported()
+        )
```

Comment:
```suggestion
        self.has_replicas = self.has_workloads 
```

And then set `has_replicas = False` on sidecars, jobs and anything that legitimately doesn't support it. You won't need to specify `values_file_path_overrides` then.